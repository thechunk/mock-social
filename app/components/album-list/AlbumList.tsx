import * as React from 'react';
import {PureComponent} from "react";
import {FlatList, GestureResponderEvent} from "react-native";
import withUserAlbumsData from "../../hoc/withUserAlbumsData";
import {ImageListCell} from "../common/list";
import withNavigationParamsAsProps from "../../hoc/withNavigationParamsAsProps";
import {NavigationInjectedProps} from "react-navigation";
import styles, {inherited} from "./styles";

class AlbumList extends PureComponent<IAlbumList & NavigationInjectedProps & IWithUserAlbumsDataProps &
    IWithUserAlbumsDataOptionalProps> {
    navigateToPhotoGrid(album: IAlbum) {
        return (event: GestureResponderEvent) => {
            this.props.navigation.navigate('PhotoGrid',
                { hydrate: album, id: album.id } as IWithUserPhotosDataOptionalProps)
        }
    }

    render() {
        return (
            <FlatList data={this.props.albums} keyExtractor={v => v.id.toString()} renderItem={v => (
                <ImageListCell
                    key={v.index}
                    id={v.item.id}
                    title={v.item.title}
                    cellStyle={[inherited.cell, styles.cell]}
                    imageStyle={styles.image}
                    titleStyle={[inherited.title]}
                    subtitleStyle={[inherited.subtitle]}
                    onPress={this.navigateToPhotoGrid(v.item)} />
            )} />
        );
    }
}

export default withNavigationParamsAsProps(withUserAlbumsData(AlbumList));