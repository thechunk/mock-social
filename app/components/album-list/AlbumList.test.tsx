import * as React from 'react';
import {AlbumList} from './AlbumList';

import {shallow} from 'enzyme';

describe('AlbumList', () => {
    const navigation = {navigate: jest.fn()};

    beforeEach(() => {
        navigation.navigate.mockClear();
    });

    it('renders correctly', () => {
        expect(shallow(<AlbumList albums={{}} loading={false} navigation={navigation} />))
            .toMatchSnapshot();
    });

    it('navigates to PhotoGrid on album tap', () => {
        const wrapper = shallow(<AlbumList albums={{}} loading={false} navigation={navigation} />);
        const album = {id: 1};
        wrapper.instance().navigateToPhotoGrid(album)();
        expect(navigation.navigate).toBeCalledWith('PhotoGrid', {hydrate: album, id: album.id});
    });
});
