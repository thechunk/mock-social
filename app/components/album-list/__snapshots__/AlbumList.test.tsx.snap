// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumList renders correctly 1`] = `
<FlatList
  contentContainerStyle={
    Object {
      "paddingBottom": 25,
      "paddingTop": 89,
    }
  }
  data={Object {}}
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  scrollIndicatorInsets={
    Object {
      "bottom": 25,
      "top": 89,
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
